ident/iterator.go

```diff
@@ -86,29 +90,60 @@ func NewStringIDsIterator(ids ...string) Iterator {
 
 // NewStringIDsSliceIterator returns a new Iterator over a slice of strings.
 func NewStringIDsSliceIterator(ids []string) Iterator {
+	// allocate a single bytes reference to not require multiple allocations as we iterate
+	idBytes := checked.NewBytes(nil, nil)
+	idBytes.IncRef()
+
 	iter := &stringSliceIter{
 		backingSlice: ids,
 		currentIdx:   -1,
+		currentID:    BinaryID(idBytes),
+		idBytes:      idBytes,
 	}
 	return iter
 }
 
 type stringSliceIter struct {
 	backingSlice []string
 	currentIdx   int
-	currentID    ID
+
+	currentID ID
+	idBytes   checked.Bytes
 }
 
 func (i *stringSliceIter) Next() bool {
 	i.currentIdx++
 	if i.currentIdx < len(i.backingSlice) {
-		i.currentID = StringID(i.backingSlice[i.currentIdx])
+		i.setCurrent(i.backingSlice[i.currentIdx])
 		return true
 	}
-	i.currentID = nil
+	i.release()
 	return false
 }
 
+func (i *stringSliceIter) setCurrent(s string) {
+	// we already hold a reference to the underlying bytes over the lifecycle
+	// of the iterator, don't need another one here.
+	i.idBytes.Reset([]byte(s))
```

Comment:
Hm (as discussed), maybe too much work to avoid extra allocs that will still ultimately require a byte slice alloc here.

Reply:
Yep that's fair. Reverted in the next version.